tests/components/fritzbox/test_config_flow.py

```diff
@@ -159,6 +182,24 @@ async def test_ssdp_not_successful(hass: HomeAssistantType, fritz: Mock):
     assert result["reason"] == "not_found"
 
 
+async def test_ssdp_not_supported(hass: HomeAssistantType, fritz: Mock):
+    """Test starting a flow from discovery with unsupported device."""
+    fritz().get_device_elements.side_effect = HTTPError("Boom")
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": "ssdp"}, data=MOCK_SSDP_DATA
+    )
+    assert result["type"] == "form"
+    assert result["step_id"] == "confirm"
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        user_input={CONF_PASSWORD: "whatever", CONF_USERNAME: "whatever"},
+    )
+    assert result["type"] == "abort"
+    assert result["reason"] == "not_supported"
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5439029)